orttraining/orttraining/python/training/execution_agent.py

```diff
@@ -107,33 +115,31 @@ def __init__(self, path_or_bytes, session_options=None, providers=None, provider
         self._training_agent = None
         self._inference_session = None
 
-        self.create_training_agent(path_or_bytes, session_options, providers, provider_options)
-
-    def create_training_agent(self, path_or_bytes, session_options, providers, provider_options):
         self._inference_session = onnxruntime.InferenceSession(path_or_bytes, session_options,
                                                                providers, provider_options)
-        self._training_agent = C_TrainingAgent(self._inference_session._sess)
+
+        self._training_agent = C_TrainingAgent(self._inference_session._sess, fw_feed_names, fw_fetches_names,
+                                               fw_outputs_device_info, bw_feed_names, bw_fetches_names, bw_outputs_device_info)
 
     def io_binding(self):
         """Return an onnxruntime.IOBinding object`."""
 
         return IOBinding(self._inference_session)
 
-    def run_forward(self, iobinding, run_options):
+    def run_forward(self, feeds, fetches, state):
         """
-         Compute the forward subgraph until it hits the Yield Op.
-         :param iobinding: the iobinding object that has graph inputs/outputs bind.
-         :param run_options: See :class:`onnxruntime.RunOptions`.
+         Compute the forward subgraph for given feeds and fetches.
+         :param feeds: Inputs to the graph run.
+         :param fetches: Outputs of the graph run.
```

Comment:
feed/fetch vs. input/output

Reply:
I think we use feeds/fetches pretty consistently, here, I'm just explaining what feed and fetches means in a more simple terms.